signer/src/stacks/contracts.rs

```diff
@@ -987,11 +1056,98 @@ impl AsContractCall for RejectWithdrawalV1 {
+            .get_storage()
+            .get_stacks_chain_tip(&req_ctx.chain_tip.block_hash)
+            .await
+            .unwrap()
+            .unwrap();
+        let maybe_report = ctx
+            .get_storage()
+            .get_withdrawal_request_report(
+                &req_ctx.chain_tip.block_hash,
+                &stacks_chain_tip.block_hash,
+                &self.id,
+                &ctx.config().signer.public_key(),
+            )
+            .await?;
+
+        let Some(report) = maybe_report else {
+            return Err(WithdrawalRejectErrorMsg::RequestMissing.into_error(req_ctx, self));
+        };
+
+        match report.status {
+            WithdrawalRequestStatus::Confirmed => (),
+            WithdrawalRequestStatus::Fulfilled(_txid) => {
+                // fails #2
+                return Err(WithdrawalRejectErrorMsg::RequestFulfilled.into_error(req_ctx, self));
+            }
+            WithdrawalRequestStatus::Unconfirmed => {
+                // fails #1
+                return Err(WithdrawalRejectErrorMsg::RequestUnconfirmed.into_error(req_ctx, self));
+            }
+        }
+
+        // 3. The request is final. Check that we've seen six new bitcoin blocks since observing
+        //    the bitcoin anchor block associated with the Stacks block confirming the withdrawal
+        //    request transaction. Fail the withdrawal request if we’ve observed less than 6
+        //    bitcoin blocks since the anchor block.
+
+        let request_block_height = report.bitcoin_block_height;
+        let blocks_observed = req_ctx.chain_tip.block_height - request_block_height;
+
+        if blocks_observed < 6 {
+            return Err(WithdrawalRejectErrorMsg::RequestNotFinal.into_error(req_ctx, self));
+        }
+
+        // 4. The request is collectively rejected or expired.
+        let signer_votes = ctx
+            .get_storage()
+            .get_withdrawal_request_signer_votes(&self.id, &req_ctx.aggregate_key)
+            .await?;
+        let signer_bitmap = BitArray::<[u8; 16]>::from(signer_votes);
+
+        if signer_bitmap != self.signer_bitmap {
+            return Err(WithdrawalRejectErrorMsg::BitmapMismatch.into_error(req_ctx, self));
+        }
+
+        let withdrawal_signers = &ctx
+            .get_storage()
+            .get_withdrawal_signers(self.id.request_id, &self.id.block_hash)
+            .await?;
+
+        let mut rejected_count = 0;
+        for signer in withdrawal_signers {
+            if !signer.is_accepted {
+                rejected_count += 1;
+            }
+        }
+
+        let threshold = req_ctx.signatures_required;
+        let num_signers = ctx.state().current_signer_set().get_signers().len();
+        let reject_threshold = num_signers - threshold as usize + 1;
+
+        if blocks_observed < WITHDRAWAL_BLOCKS_EXPIRY && rejected_count < reject_threshold {
+            return Err(WithdrawalRejectErrorMsg::RequestNotRejected.into_error(req_ctx, self));
+        }
```

Comment:
If we go with what is in https://github.com/stacks-network/sbtc/issues/1363, we can simplify this to just, and get rid of the other stuff around vote counting.
```rust
if blocks_observed < WITHDRAWAL_BLOCKS_EXPIRY {
    return Err(WithdrawalRejectErrorMsg::RequestNotFinal.into_error(req_ctx, self));
}
```

Reply:
Done.